src/lib/util.ts

```diff
@@ -292,20 +292,47 @@ export function transformDiscordLocale(discordLocale: string): string {
 	else return "en"
 }
 
-export function updateButtonColors(row: MessageActionRow, page: number, pages: unknown[]) {
-	if (page === 0) {
-		row.components.forEach(button => {
-			if (button.customId === "first" || button.customId === "previous") (button as MessageButton).setStyle("SECONDARY").setDisabled(true)
-			else (button as MessageButton).setStyle("SUCCESS").setDisabled(false)
-		})
-	} else if (page === pages.length - 1) {
-		row.components.forEach(button => {
-			if (button.customId === "last" || button.customId === "next") (button as MessageButton).setStyle("SECONDARY").setDisabled(true)
-			else (button as MessageButton).setStyle("SUCCESS").setDisabled(false)
-		})
-	} else row.components.forEach(button => (button as MessageButton).setStyle("SUCCESS").setDisabled(false))
-
-	return row
+export function createButtonControls(
+	pageIndex: number,
+	pages: unknown[],
+	options: { localizations?: GetStringFunction; itemName?: string } = { itemName: "page" },
+) {
+	const isLast = pageIndex === pages.length - 1,
+		disabledStyle = (disabled: boolean) => (disabled ? "SECONDARY" : "SUCCESS"),
+		isFirst = pageIndex === 0
+
+	return new MessageActionRow({
+		components: [
+			new MessageButton({
+				style: disabledStyle(isFirst),
+				emoji: "⏮️",
+				customId: "first",
+				label: options.localizations ? options.localizations("pagination.first") : `First ${options.itemName}`,
+				disabled: isFirst,
+			}),
+			new MessageButton({
+				style: disabledStyle(isFirst),
+				emoji: "◀️",
+				customId: options.localizations ? options.localizations("pagination.first") : `Last ${options.itemName}`,
+				label: "Previous log",
```

Comment:
This is wrong, the custom ID is previous, the label is localized. Should use the same method suggested above